src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -80,26 +81,38 @@ private static boolean isStreamingEnabled(KustoSinkConfig config) throws JSONExc
         return false;
     }
 
-    public static Client createKustoEngineClient(KustoSinkConfig config) {
-        try {
-            String engineClientURL = config.getKustoEngineUrl();
-            if (!Strings.isNullOrEmpty(config.getAuthAppid())) {
-                if (Strings.isNullOrEmpty(config.getAuthAppkey())) {
-                    throw new ConfigException("Kusto authentication missing App Key.");
-                }
-                ConnectionStringBuilder kcsb = ConnectionStringBuilder.createWithAadApplicationCredentials(
-                        engineClientURL,
-                        config.getAuthAppid(),
-                        config.getAuthAppkey(),
-                        config.getAuthAuthority());
-                kcsb.setClientVersionForTracing(Version.CLIENT_NAME + ":" + Version.getVersion());
+    public static ConnectionStringBuilder createKustoEngineConnectionString(KustoSinkConfig config, String clusterUrl) {

```

Comment:
Extracted the logic to create a connection string builder inside this method. The `clusterUrl` allows to use the same method for both the Ingest and Streaming use cases.